store/tikv/region_cache.go

```diff
@@ -472,6 +500,63 @@ func (c *RegionCache) loadRegionByID(bo *Backoffer, regionID uint64) (*Region, e
 	}
 }
 
+// scanRegions scans at most `limit` regions from PD, starts from the region containing `startKey` and in key order.
+func (c *RegionCache) scanRegions(bo *Backoffer, startKey []byte, limit int) ([]*Region, error) {
+	if limit == 0 {
+		return nil, nil
+	}
+
+	var backoffErr error
+	for {
+		if backoffErr != nil {
+			err := bo.Backoff(BoPDRPC, backoffErr)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+		metas, leaders, err := c.pdClient.ScanRegions(bo.ctx, startKey, limit)
+		if err != nil {
+			tikvRegionCacheCounterWithScanRegionsError.Inc()
+		} else {
+			tikvRegionCacheCounterWithScanRegionsOK.Inc()
+		}
+		if err != nil {
+			backoffErr = errors.Errorf(
+				"scanRegion from PD failed, startKey: %q, limit: %q, err: %v",
+				startKey,
+				limit,
+				err)
+			continue
+		}
+		if len(metas) == 0 {
+			return nil, errors.New("PD returned no region")
+		}
+		if len(metas) != len(leaders) {
+			return nil, errors.New("PD returned mismatching region metas and leaders")
+		}
+		regions := make([]*Region, 0, len(metas))
+		for i, meta := range metas {
+			region := &Region{
+				meta: meta,
+				peer: nil,
+			}
+			leader := leaders[i]
+			if leader.GetId() != 0 {
+				region.SwitchPeer(leader.GetStoreId())
+			}
+			regions = append(regions, region)
+		}
+		if len(regions) == 0 {
```

Comment:
Duplicate condition checking with L531.

Reply:
It's not the same, since not every item in `metas` will be added to `regions`